### PR TITLE
4111 performance improvements for common files search from tree expansion listener

### PR DESCRIPTION
**Pull request by aziehl**
PR for review of improvements, per 4111

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/InstanceCountNode.java [hunk 1]
```diff
@@ -19,62 +19,92 @@
...
      * Number of matches found for each of the MD5 children.
+     *
      * @return int match count
      */
     int getInstanceCount() {
         return this.instanceCount;
     }
 
+    public void refresh() {
+        new SwingWorker<Void, Void>() {
```
[the remark this answers is not in the document]
Comment (aziehl) [reply Core/src/org/sleuthkit/autopsy/commonfilesearch/InstanceCountNode.java]: Good point, I'd forgotten that the async boolean switch existed when I added the SwingWorker. Incorporating that change.

File: Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java [hunk 2]
```diff
@@ -102,6 +103,20 @@
     public DataResultViewerTable() {
         this(null, Bundle.DataResultViewerTable_title());
     }
+    /**
```
[the remark this answers is not in the document]
Comment (aziehl) [reply Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: Agreed, we could add the listener directly in CommonAttributesSearchResultsViewerTable if the outlineView was protected. At the time I felt adding a public constructor had less impact than increasing the visibility of a class member, but if there's other uses for it, we should go that route.

[the remark this answers is not in the document]
Comment (aziehl) [reply Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: I didn't make the outlineView protected yet, do you think it's fine do to so?

[the remark this answers is not in the document]
Comment (esaunders) [reply Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: I don't know if there are other uses for it at this point but we certainly have your use case where you have subclassed DataResultViewerTable. It seems as though the outlineView member is the gateway through which a lot of the behavior of the result viewer can be tweaked so I don't have an issue with exposing it to subclasses.

**Comment by aziehl** [comment]
Addressed @esaunders feedback and pushed the changes.

File: Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (aziehl) [reply Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: Updated, with the change.

**Comment by esaunders** [comment]
Looks like this PR now has conflicts. @aziehl Can you take care of them?

**Comment by aziehl** [comment]
Pushed merge conflict fixes.

**Pull request merged**